nre/bpf/nodefw/nodefw-ebpf/src/main.rs

```diff
@@ -0,0 +1,114 @@
+#![no_std]
+#![no_main]
+
+use aya_bpf::{
+    bindings::xdp_action,
+    macros::{map, xdp},
+    maps::HashMap,
+    programs::XdpContext,
+};
+use aya_log_ebpf::info;
+use core::mem;
+// TODO see if this is preferred over ptr_at
+// use memoffset::offset_of;
```

Comment:
supposed to be more mem efficient apparently. I only found one example use of it 
`let destination = u32::from_be(ctx.load(offset_of!(iphdr, daddr))?);` this [example](https://github.com/aya-rs/book/blob/a0089f83de5684d189cfd55422b8032b138fef8d/examples/cgroup-skb-egress/cgroup-skb-egress-ebpf/src/main.rs#L46) is using skb though.

probably an optimization we could try later.

Reply:
yeah i wasn't exactly sure how to use it yet. the way they refer to fields in the packet have changed since i used aya last. it's on the TODO.